crates/nri-resctrl-plugin/src/lib.rs

```diff
@@ -257,6 +438,90 @@ impl<P: FsProvider + Send + Sync + 'static> Plugin for ResctrlPlugin<P> {
 mod tests {
     use super::*;
     use protobuf::SpecialFields;
+    use tokio::time::{timeout, Duration};
+
+    #[derive(Clone, Default)]
+    struct TestFsState {
+        files: std::collections::HashMap<std::path::PathBuf, String>,
+        dirs: std::collections::HashSet<std::path::PathBuf>,
+        no_perm_files: std::collections::HashSet<std::path::PathBuf>,
+    }
+
+    #[derive(Clone, Default)]
+    struct TestFs {
+        state: std::sync::Arc<std::sync::Mutex<TestFsState>>,
+    }
+
+    impl TestFs {
```

Comment:
are we re-implementing the mock filesystem that exists in the resctrl tests? Why not have a single shared implementation? can separate it into a module in resctrl (that gets compiled when compiling tests). If we move to have a PID source provider here that uses cgroups_rs instead of the filesystem, I think we don't need the fs interface here -- only in resctrl anyway (we need it to control tests here however)